openapi_python_client/templates/property_templates/date_property.pyi

```diff
@@ -1,6 +1,11 @@
 {% macro construct(property, source, initial_value="None") %}
 {% if property.required %}
+{% if property.nullable %}
+{{ property.python_name }} = {{ source }}
+{{ property.python_name }} = isoparse({{ property.python_name }}).date() if {{ property.python_name }} else None
```

Comment:
The changes as they are here would cause type issues, I think, because the type of `{{property.python_name}}` changes from that of `{{source}}` (`Optional[str]`) to `Optional[date]`

You would need to put the `{{source}}` in a temp variable for this implementation:

```suggestion
_{{ property.python_name }} = {{ source }}
{{ property.python_name }} = isoparse(_{{ property.python_name }}).date() if {{ property.python_name }} else None
```


But I think @dbanty's suggestion would work and would be a bit simpler.

Reply:
Alright, I will try to fix that today!